Blossoming Pots Guidebook

runcows opened the pull request:
This PR for Blossoming Pots
- adds Guidebook support
- updates the Module description in `beet.yaml` 
- rewrites `README.md`
Module description and `README.md`  changes are reflected on the wiki.

Would like feedback on wording of things as I'm not sure on if it sounds normal or super weird.

Also would like some feedback on the advancement trigger for guidebook. I used beehive_inspector as the template for the guidebook file, which uses `minecraft:inventory_changed`. I changed it to test for decorated pot or flower pot. It might make more sense to use a different trigger, but I'm not sure.

catter1 commented on gm4_blossoming_pots/data/gm4_blossoming_pots/guidebook/blossoming_pots.json:
```diff
@@ -0,0 +1,46 @@
+            },
+            {
+              "translate": "text.gm4.guidebook.blossoming_pots.description",
+              "fallback": "Interacting with a Decorated Pot or empty Flower Pot while holding any one of the supported items will plant it in the pot.\n\nCheck the full supported item list on the wiki."
```
I think it might sound better if it was worded something something along the lines of:

"To pot a plant, interact with a Decorated Pot or empty Flower Pot while holding a supported item."

English fails me here, it would feel more natural if "pot" and "plant" weren't both the verbs and nouns :(

runcows replied:
English fails us all. 😔 But yes I like your wording a lot better, I will steal it!

catter1 commented on gm4_guidebook/triggers.json:
```diff
@@ -1,6 +1,6 @@
 {
   "__important__": "Generated by generate_guidebooks.py. Don't manually update this",
```
Isn't this not supposed to be updated? Or was it done via the script?

BPR02 replied:
It looks like how the script would update it, so it should be fine

catter1 commented on gm4_blossoming_pots/beet.yaml:
```diff
@@ -21,7 +21,7 @@ meta:
       schedule_loops: [main]
     website:
-      description: Decorated Pots and Flower Pots extended to allow all kinds of things to be placed in them! Decorate in Style!
+      description: Greatly expands the selection of potable plants. Use Decorated Pots as just abnormally large flower pots. Decorate in Style!
```
It sounds better to me by removing the word "just" in the second sentence.

catter1 replied:
If you make this change, make sure to reflect it in the README too (line 3).

BPR02 replied:
Also in translations.csv (line 2)

BPR02 replied:
Could change it to "Simply use decorated pots as abnormally large flower pots" if you wanna remove the word "just" but keep the idea of it being straight forward to use.

runcows replied:
Either one could work, but I think I like it better by removing "just". It feels vaguely like an early 2000s american informercial that way, which I think is funny

catter1 commented on gm4_blossoming_pots/README.md:
```diff
@@ -1,26 +1,11 @@
-  - Chorus Flowers
-  - All Amethyst Buds & Clusters
-  - Dripstone
+- Interacting with a decorated pot or empty flower pot while holding any one of the supported items will plant it in the pot.
```
In the guidebook (and in a couple other places in this same file), you capitalised "decorated pot" and "flower pot", but it is lowercase here. I don't know what the standard is, but I assume it should be the same between both.

catter1 commented on gm4_blossoming_pots/README.md:
```diff
@@ -1,26 +1,11 @@
-  - All Amethyst Buds & Clusters
-  - Dripstone
+- Interacting with a decorated pot or empty flower pot while holding any one of the supported items will plant it in the pot.
+- Interacting with a flower pot from this module with an empty hand will empty the flower pot, just like vanilla flower pots. Decorated Pots will have to be broken to unplant that which you have potted.
```
The second sentence definitely feels weird. I would recommend something such as:
- "To empty a Decorated Pot of its plant, it must be broken."
- "A Decorated Pot must be broken to have its plant removed."
- "Break a Decorated Pot to remove the plant."

runcows replied:
Thank you I will take that first one, and also tweak the flower pot sentence just before it

catter1 reviewed (changes requested):
Potential wording suggestions. Accept what you like. I'm also not sure on how standard the wording/tenses/voices are across all module descriptions, so take that into consideration as well.

BPR02 commented on gm4_blossoming_pots/README.md:
```diff
@@ -1,26 +1,11 @@
+- Interacting with a decorated pot or empty flower pot while holding any one of the supported items will plant it in the pot.
+- Interacting with a flower pot from this module with an empty hand will empty the flower pot, just like vanilla flower pots. Decorated Pots will have to be broken to unplant that which you have potted.
+- Decorated pot plants often have multiple stages for a plant.
+- [Click Here](https://wiki.gm4.co/Blossoming_Pots#Supported_Items) to see the full list of supported items
```
I'd make the hyperlink text say "Read the wiki" instead of "Click Here" so it's apparent where it'll take you before you click

runcows replied:
Ah, normally I would have done that, but I was basing the hyperlink here off of the better_armour_stands README which i had randomly clicked over to looking at other modules files. 👍 If you want, I can go through other modules and make a PR to change hyperlinks to that style, if there's others like it

runcows replied:
Yeah, I can confirm that no other modules say "Click here" in the hyperlink text (if there is one), except for better_armour_stands. And there seems to be a style standard for linking to the wiki across other `README`s that armour stands just doesn't use. It's like I blind folded shot a dart at a wall and managed to hit the one thing that wasn't a target :D

BPR02 commented on gm4_blossoming_pots/data/gm4_blossoming_pots/guidebook/blossoming_pots.json:
```diff
@@ -0,0 +1,46 @@
+      }
+    ]
+  }
+  
```
missing newline at end of file
```suggestion
  
```

runcows replied:
Github web is telling me "This suggestion is invalid because no changes were made to the code." and won't let me commit the suggestion :(

BPR02 commented on gm4_blossoming_pots/data/gm4_blossoming_pots/guidebook/blossoming_pots.json:
```diff
@@ -0,0 +1,46 @@
+        }
+      }
+    },
+    "sections": [
```
You may want to consider adding the list of pottable plants as another section (or the same one). The goal is for them to learn everything using the guidebook instead of the wiki page (some guidebook entries don't follow this tho, so it's fine either way).

BPR02 commented on gm4_blossoming_pots/translations.csv:
```diff
@@ -0,0 +1,3 @@
+key,en_us
+text.gm4.guidebook.module_desc.blossoming_pots,Greatly expands the selection of potable plants. Use Decorated Pots as just abnormally large flower pots. Decorate in Style!
+text.gm4.guidebook.blossoming_pots.description,"Interacting with a Decorated Pot or empty Flower Pot while holding any one of the supported items will plant it in the pot.\n\nCheck the full supported item list on the wiki."
```
As catter said in another comment, this should have consistent capitalization as the rest of the instances of this text.

BPR02 reviewed (commented):
Looks like the correct way to implement guidebook pages, so as long as it works in-game the script should be working. I left a few comments regarding the phrasing of certain things.

runcows commented:
Greatly appreciate the suggestions, I implemented I think all of the wording changes suggested by Beeps and Catter. Going to look into adding a guidebook section for pottable plants / supported items.

runcows commented:
I added a condensed supported item list as 2 additional pages to the guidebook and added a sentence on removing plants to the first page. 

This is the condensed item list I came up with. Grouping all of the items normally plantable in vanilla flower pots together helped condense this list considerably. 

Supported Items:
- Vanilla Flower Pot plants
- Pale hanging moss
- Pointed dripstone
- Amethyst crystals
- Leaves
- Short & tall grass
- Lush cave plants
- Sugar cane
- Nether vines
- 2 tall plants
- Chorus flower
- Non-sniffer seeds
- Nether wart
- Sweet berries
- Coral reef plants

TheThanathor commented:
Tested this in game and found no problems, pages unlock when expected and look good
